singleapplication_p.cpp

```diff
@@ -328,10 +328,12 @@ void SingleApplicationPrivate::slotConnectionEstablished()
         }
     );
 
-    QObject::connect(nextConnSocket, &QLocalSocket::disconnected,
-        [nextConnSocket, this](){
+    QObject::connect(nextConnSocket, &QLocalSocket::disconnected, nextConnSocket, &QLocalSocket::deleteLater);
+
+    QObject::connect(nextConnSocket, &QLocalSocket::destroyed,
+        [nextConnSocket, this](QObject *obj){
+            Q_UNUSED(obj)
```

Comment:
In Qt, you don't need a slot to have the same number of arguments as a signal. So, you can just omit `QObject *`.

```suggestion
        [nextConnSocket, this](){
```

Reply:
Did not know that! Thanks!